### PR TITLE
Deprecate sort option reverse.

### PR DESCRIPTION
**Pull request by MaineC**
Replace with "order" instead.

Relates to #17047

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortParser.java [hunk 1]
```diff
@@ -57,6 +58,8 @@
  */
 public class GeoDistanceSortParser implements SortParser {
 
+    private static final ParseField REVERSE_FIELD = new ParseField("reverse").withAllDeprecated("order");
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortParser.java]: I'm not sure this works as expected, we are not replacing "reverse" with "order", aren't we? That would mean we need to accept `"order" : true` now, which I think we don't. I'd put some note here like "replaced by order" like we do in the RangeQueryParse here.

**Comment by cbuescher** [comment]
@MaineC took a quick look, can you also remove the mention of "reverse" in `docs/reference/search/request/sort.asciidoc` and maybe we shouldn't use it in the `toXContent` of the builders (ScriptSortBuilder and GeoDistanceSortBuilder) any more but use "order" there instead since the internal field and the setters are already called "order".

**Comment by MaineC** [comment]
@cbuescher Makes sense. Thanks for catching this. Changed.

**Comment by cbuescher** [comment]
@MaineC thanks, did you push any changes re the `withAllDeprecated` comment I made above?

**Comment by MaineC** [comment]
> did you push any changes re the withAllDeprecated comment I made above

Thanks for the reminder - completely missed that

**Comment by MaineC** [comment]
@cbuescher Looks better now?

**Comment by cbuescher** [comment]
Sorry, I missed mentioning that the same comment applies to the parse fields in ScriptSortParser and the SortParseElement. Can you add that change there as well?

**Comment by MaineC** [comment]
Sorry, should have been obvious. Fixed.

**Comment by cbuescher** [comment]
Great, LGTM

**Comment by cbuescher** [comment]
@MaineC has this been merged already and can be closed?

**Comment by MaineC** [comment]
@clintongormley Do we plan to do a 2.4 release (that is, is it enough to merge this to 2.x or should this rather be merged to the 2.3 branch)?

**Comment by clintongormley** [comment]
@MaineC this can go to 2.3 as well

**Pull request merged**